### ISSUE TITLE
Why I see two extra service check calls in my every request? Slowing down a lot

### ISSUE THREAD

**Issue by zonito** [cloudendpoints/endpoints-python#140]
![screenshot 2018-02-27 00 18 18](https://user-images.githubusercontent.com/2670622/36681535-d4cc2fae-1b53-11e8-869e-661e484c0f06.png)

I have attached the screenshot of the trace, Where I see 2 urlfetch calls which I didn't make and it is slowing down my all API calls.

And somehow my AppEngine project with the endpoint, for just simple operation, all requests are taking more than 1s. 

![screenshot 2018-02-27 00 19 53](https://user-images.githubusercontent.com/2670622/36681712-615baf30-1b54-11e8-8c17-88f8b41ae5e5.png)

Please share your feedback on how to optimize it? 

Thanks

More info:
- Using endpoint v2
- Region: asia-south
- Appengine: Standard
- Runtime: python27

**Comment by tangiel** [comment]
What scaling type are you using?

**Comment by zonito** [comment]
Autoscaling

**Comment by tangiel** [comment]
This is why. If you use basic scaling, those requests will take place in a background thread. If you don't need the Endpoints features in Cloud Console, you should be able to disable this by removing the Endpoints environment variables in `app.yaml`.

**Comment by zonito** [comment]
@tangiel When I remove endpoint configuration from app.yaml - Looks like things are on normal track. But I want to use endpoint feature to generate android SDK, also I received this email from Google for deprecation of v1. Will this impact me?

```
We’re writing to remind you that Cloud Endpoints Frameworks v1 is deprecated as of August 2, 2017. Please migrate to Cloud Endpoints Frameworks v2 at your earliest convenience. The v1 Endpoints Frameworks will cease to serve traffic on August 2, 2018. We have easy-to-follow migrations for both our Java and Python frameworks that will allow you to upgrade your API to the new versions without disrupting your API clients. Very importantly, your existing clients will work without redeployment.
```

Thanks for the guidance. Much appreciated.

**Comment by zonito** [comment]
It works even without basic_scaling / with basic scaling - works same. So I will stick to autoscale as of now. I am a bit afraid of depreciation of v1.

**Comment by tangiel** [comment]
If you're just migrating, you don't need the app.yaml configuration to continue generating client libraries.

**Comment by zonito** [comment]
Okay, removing ENV_VARIABLES of endpoints - is it still `v2`?

And I see the different issue now:

![screenshot 2018-02-27 08 48 32](https://user-images.githubusercontent.com/2670622/36704400-2ba03be4-1b9b-11e8-84ba-6937588a8b8b.png)

I have created 2 handlers:
1. Using endpoints
2. Without endpoints

The difference you will see it in above screenshot, internal code is same but the response time/latency is very high in using endpoints. Can you please help me to reduce this as well?

Thanks

**Comment by tangiel** [comment]
Yes, it's still v2, just with API Management disabled. If this is your latency after removing the config, I'll probably delegate to @inklesspen for investigation. You could try a higher instance size to see if it would help, though it obviously would not be ideal.

**Comment by zonito** [comment]
Thanks @tangiel for your help.

In my api call, i am just reading few records from datastore (max 10 records). Just for this increasing the instance class wont make sense.

@inklesspen please guide

**Comment by tangiel** [comment]
I was just suggesting raising it to see if it helps--you don't need to keep it at a high level. It would be good to know.

**Comment by inklesspen** [comment]
@zonito Can you provide trace data of the two different requests (like you did in your opening comment)? Thanks.

**Comment by inklesspen** [comment]
I created a very barebones API, just a couple of methods, and all they do is spit out some data from the environment. When deployed with API management, requests take around 150 ms to process; when deployed without API management, they take around 5 ms.

I don't know where the overhead is coming from in your case. I can think of some possibilities, but it's really difficult to say without examining your code.

I'd be interested to see if you get different numbers when either (a) replacing the implementation of the method with something that just spits out a fixed result, or (b) deploying an API that contains _just_ that method and nothing else.

**Comment by zonito** [comment]
Okay, I have sent you an email at jon@inklesspen.com

To give more insight, just in case, I am using `asia-south` region.

**Comment by inklesspen** [comment]
I'm afraid I can't see any obvious causes of the slowdown. That said, there is some overhead involved in the Endpoints Framework, but I don't understand how you'd get 400 ms of overhead.

I hope to introduce better profiling support in a future version of the framework.

**Comment by zonito** [comment]
What to do meanwhile? This will end up creating many instances if load increases - request generally take 100ms but now 500ms (4 times higher).

**Comment by inklesspen** [comment]
@zonito Please see the recently-released version 4.0.0, which should greatly improve your request handling time.
